cmd/poller/poller.go

```diff
@@ -281,13 +285,69 @@ func (p *Poller) Init() error {
 	}
 	logger.Debug().Msgf("set poller schedule with %s frequency", pollerSchedule)
 
+	// ASUP parameters will be loaded from Tools section
+	// since this feature will probably be delegated to a separate tool
+	toolsParams, err := conf.GetTools(p.options.Config)
+	if err != nil {
+		logger.Error().Stack().Err(err).Msg("load Tools config")
+		return err
+	}
+
+	if toolsParams.AsupEnabled != nil && *toolsParams.AsupEnabled {
+		if !p.targetIsOntap() {
+			logger.Info().Msg("disabling ASUP messaging as target is not ONTAP")
+		} else if err = p.schedule.NewTaskString("asup", asupInterval, p.startAsup); err != nil {
+			logger.Error().Stack().Err(err).Msg("set ASUP interval:")
+			return err
+		} else {
+			logger.Info().Msgf("set ASUP interval to %s", asupInterval)
+
+			// Make sure harvest instance has a UUID, if not create one and safe for later use
+			// @TODO: race conditions can occur here, so maybe do this at install time?
+			if toolsParams.HarvestUUID != nil {
+				logger.Info().Msg("parsing harvestUUID from config...")
+				if harvestUUID, err = uuid.Parse(*toolsParams.HarvestUUID); err != nil {
+					logger.Error().Stack().Err(err).Msg("parse stored harvest UUID")
+				}
+				logger.Info().Msgf("using stored harvest UUID [%s]", *toolsParams.HarvestUUID)
+			} else if harvestUUID, err = uuid.NewRandom(); err != nil {
+				logger.Error().Stack().Err(err).Msg("generate harvest UUID")
+				return err
+			} else {
+				s := harvestUUID.String()
+				toolsParams.HarvestUUID = &s
+				logger.Info().Msgf("generated harvest UUID [%s]", *toolsParams.HarvestUUID)
+				// Safe UUID
+				if err = conf.SafeHarvestConfig(p.options.Config); err != nil {
+					logger.Error().Stack().Err(err).Msg("safe config")
+					return err
+				} else {
+					logger.Info().Msgf("saved harvest UUID [%s] to new_%s", harvestUUID.String(), p.options.Config)
+				}
+			}
+		}
+
+	} else {
+		logger.Info().Msg("disabling ASUP messaging, by preference")
+	}
+
 	// famous last words
 	logger.Info().Msg("poller start-up complete")
 
 	return nil
 
 }
 
+// to the asup
+func (p *Poller) startAsup() (*matrix.Matrix, error) {
```

Comment:
is the return correct? looks like we only ever return `nil` for the Matrix - should this method be `func (p *Poller) startAsup() error {` instead?

Reply:
This function is invoked by scheduler and it's signature should be func name() (*matrix.Matrix, error). That's the reason for this return.